### PR TITLE
Update dependencies and tests

### PR DESCRIPTION
**Pull request by khaledh**
This PR addresses #103:
- Updates dependencies to more recent versions; most notably:
  - Update Python from 3.6 to >=3.7
  -  Update `pyspark` from 2.4.1 to 3.x
  - Update the rest of dependencies to their most recent version
- Fix tests, mostly by adopting a more recent representation of stringified schema

✅  All tests pass locally.

### CODE REVIEW

**Review by mattjw — COMMENTED** [review]
Thank you very much for the PR! 🙇‍♂️ sparkql is certainly in need of some updates. I'm away from my laptop currently, but I look forward to taking a proper look tomorrow.

**Comment by mattjw** [comment]
The old CI will need updating. I've made a start here https://github.com/mattjw/sparkql/pull/105 (for testing). I'll come back to it in the next few days. Sorry for the wiat!

**Comment by mattjw** [comment]
@khaledh good news. I now have CI working and a passing build https://github.com/mattjw/sparkql/pull/105 I'll take some time tomorrow to merge the fixes in to main, and then I can at last get your PR merged.

BTW, the CI runner is only checking Java 11, Python 3.10.1, and PySpark 3.3.2. Sadly I don't currently have any ongoing PySpark 3.x projects to try sparkql with in a representative way, so if you have any suggestions on what it'd be best to be testing against, let me know. (There's no matrix runner at the moment, but it's something I could add, to check against a few versions.)

**Comment by khaledh** [comment]
@mattjw That's great news! I'm currently using sparkql with Python 3.10 and Spark 3.3.2/Java 11, and everything works as expected. For other versions you're right, that will require something like [tox](https://tox.wiki/en/latest/).

File: tests/unit/test_formatters.py [hunk 1]
```diff
@@ -24,40 +24,50 @@ class TestPrettySchema:
...
+    @staticmethod
+    def diff_strings(a, b):
+        print('diff strings')
+        for i, (a, b) in enumerate(zip(a, b)):
+            if a != b:
+                print(f"diff at index {i}")
+                print(f"expected: {ord(a)}")
+                print(f"actual: {ord(b)}")
+                break
+
```
Comment (mattjw) [inline tests/unit/test_formatters.py:70]:
```suggestion

```

Might be a leftover from some local testing that can be deleted?

**Review by mattjw — COMMENTED** [review]
All looks great tome. Just one small suggestion, regarding `diff_strings`.

I did some additional fixes (CI automation, and also appeasing changes in pylint version(s)) on  https://github.com/khaledh/sparkql/pull/1/files, and the project tests and builds correctly both on my local and via CI.

I've you're happy to merge https://github.com/khaledh/sparkql/pull/1 then the rest is over me. (And very straightforward.) 🚀

**Comment by mattjw** [comment]
> For other versions you're right, that will require something like [tox](https://tox.wiki/en/latest/).

Neat! I will take a look 👀 .

File: tests/unit/test_formatters.py (hunk 1 above)
Comment (mattjw) [reply tests/unit/test_formatters.py:70]: Fixed in https://github.com/khaledh/sparkql/pull/1/commits/d794f532b5162b6ce85313fde558262eadba1c60, so this change will be subsumed by https://github.com/khaledh/sparkql/pull/1

**Comment by khaledh** [comment]
Amazing! I just merged https://github.com/khaledh/sparkql/pull/1, so you should be good to go.

**Comment by mattjw** [comment]
I have a feeling that the automated release pipeline will break – but I'd kinda expected it. Won't take too long to fix. I'll add an update on this PR once it's out.

**Comment by khaledh** [comment]
Looks like v0.0.6 was published to PyPI! 🎉 

@mattjw Thank you so much for your guidance on getting this shipped 🙇🏼

**Pull request merged**